app/services/build_runner_service.rb

```diff
@@ -44,7 +44,7 @@ def build_runner_task_queue
     def find_build_runner(project_id:, sha: nil, build_number: nil)
       return build_runners.find do |build_runner|
         build_runner.project.id == project_id &&
-        (build_runner.sha == sha || build_runner.current_build.number == build_number)
+        (build_runner.current_build.sha == sha || build_runner.current_build.number == build_number)
```

Comment:
Why access `.current_build` directly? Do you think it's the better approach? If so, should we remove the `.sha` helper method in the other build_runner?

Reply:
This was a bug. I don't believe we set the sha on the build_runner, when we `prepare_build_object`. This caused `find_build_runner` to match against ANY build runner with the same `project_id` only (since `build_runner.sha == nil` and `sha == nil`)

Reply:
You'll notice that `sha` is an attribute on `BuildRunner` independent of the Build. It was possible for them to by out of sync.